src/hooks/api.tsx

```diff
@@ -91,22 +117,115 @@ export function useGovernorCouncil(
 export function useProposals(
   governorAddress: string | undefined,
   currentBlock: number | undefined,
-  enabled: boolean = true,
+  enabled: boolean = true
 ): UseQueryResult<Proposal[]> {
   const { network } = useWallet();
-  const paramsDefined = governorAddress !== undefined && currentBlock !== undefined;
+  const paramsDefined =
+    governorAddress !== undefined && currentBlock !== undefined;
 
   async function loadProposals(): Promise<Proposal[]> {
     if (paramsDefined) {
-      const proposals = await fetchProposalsByGovernor(governorAddress);
+      const proposals = (
+        await fetchProposalsByGovernor(governorAddress)
+      ).reverse();
       const updatedProposals: Proposal[] = [];
+      let proposalsToCheck = 5;
+      let nextProposalId = (await getNextPropId(network, governorAddress)) ?? 0;
+      let id = proposals[0] ? proposals[0].id : nextProposalId - 1;
+
+      // fetch proposals from ledger if not available in indexer
+      for (let i = id; i < nextProposalId - 1; i++) {
```

Comment:
In the event proposals.len() === 0 and next proposal id !== 1 this logic won't work.

Consider:
```js
let lastProposalId = (await getNextPropId ?? 0) - 1;
let lastIndexedProp = proposals.len() === 0 ? -1 : proposals[0];

if (lastProposalId === -1 && lastIndexedProp === -1) return []; // no proposals created

for (const prop_id = lastIndexedProp + 1; prop_id <= lastProposalId; prop_id++) {
    let prop = getProposal()
    if (prop !== undefined) {
        proposals.unshift(prop); // need to push to front of array. This is inefficient but we don't expect this to happen
    }
}
```